client/landing/stepper/declarative-flow/internals/steps-repository/plans/index.tsx

```diff
@@ -6,24 +6,26 @@ import {
 } from '@automattic/onboarding';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
 import PlansWrapper from './plans-wrapper';
-import type { ProvidedDependencies, Step } from '../../types';
+import type { Step } from '../../types';
 import type { MinimalRequestCartProduct } from '@automattic/shopping-cart';
 
 /**
  * @deprecated Use `unified-plans` instead. This step is deprecated and will be removed in the future.
  */
-const plans: Step = function Plans( { navigation, flow } ) {
+const plans: Step< {
+	submits: {
+		plan: MinimalRequestCartProduct | null;
+		goToCheckout: boolean;
+	};
+} > = function Plans( { navigation, flow } ) {
 	const { goBack, submit } = navigation;
 
 	const handleSubmit = ( plan: MinimalRequestCartProduct | null ) => {
-		const providedDependencies: ProvidedDependencies = {
+		const providedDependencies = {
 			plan,
+			goToCheckout: isDomainUpsellFlow( flow ) || isStartWritingFlow( flow ),
```

Comment:
This is a logic change to make typing easier.

Reply:
yes but totally harmless